Fails to connect to Sonarqube server signed by corporate PKI

Hi,

We've an internal sonarqube server, that is using a certificate signed by corporate PKI.
Action doesn't want to connect to it.

> org.sonarsource.scanner.api.internal.ScannerException: Unable to execute SonarScanner analysis
> Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

I had to fork the action repo and add following lines to the Dockerfile
```lang=Dockerfile
# Add certificate in java keystore
COPY sonarqube.corp.net.pem .
RUN keytool -keystore /etc/ssl/certs/java/cacerts -storepass changeit -noprompt -trustcacerts -importcert -alias sonarqube -file sonarqube.corp.net.pem
```

There should be an option to deal with the issue (use cert store of the host, accept as arg the hash of the certificate to trust...). It may be possible to solve that with java arg, but if it's the case it may be great to add it in the Readme.

Thanks

--- a/README.md
+++ b/README.md
@@ -63,7 +63,7 @@ If your source code file names contain special characters that are not covered b
         LC_ALL: "ru_RU.UTF-8"
 ```
 
-If your SonarQube server users a self-signed certificate, you can pass a root certificate (in PEM format) to use for validation to the java certificate store:
+If your SonarQube server uses a self-signed certificate, you can pass a root certificate (in PEM format) to the java certificate store:
 
 ```yaml
     - name: SonarQube Scan

--- a/README.md
+++ b/README.md
@@ -63,7 +63,7 @@ If your source code file names contain special characters that are not covered b
         LC_ALL: "ru_RU.UTF-8"
 ```
 
-If your sonarqube server users a self-signed certificate, you can pass a root certificate (in PEM format) to use for validation to the java certificate store:
+If your SonarQube server users a self-signed certificate, you can pass a root certificate (in PEM format) to use for validation to the java certificate store:
 
 ```yaml
     - name: SonarQube Scan

--- a/README.md
+++ b/README.md
@@ -63,6 +63,17 @@ If your source code file names contain special characters that are not covered b
         LC_ALL: "ru_RU.UTF-8"
 ```
 
+If your sonarqube server users a self-signed certificate, you can pass a root certificate (in PEM format) to use for validation to the java certificate store:
+
+```yaml
+    - name: SonarQube Scan
+      uses: sonarsource/sonarqube-scan-action@master
+      env:
+        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+        SONAR_ROOT_CERT: %{{ secrets.SONAR_ROOT_CERT }}
+```
+
 You can change the analysis base directory by using the optional input `projectBaseDir` like this:
 
 ```yaml
@@ -91,6 +102,7 @@ More information about possible analysis parameters can be found in [the documen
 
 - `SONAR_TOKEN` – **Required** this is the token used to authenticate access to SonarQube. You can read more about security tokens [here](https://docs.sonarqube.org/latest/user-guide/user-token/). You can set the `SONAR_TOKEN` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
 - `SONAR_HOST_URL` – **Required** this tells the scanner where SonarQube is hosted. You can set the `SONAR_HOST_URL` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
+- `SONAR_ROOT_CERT` – This is used to pass the scanner a custom root certificate (in PEM format), that should be used for validating the sonarqube server certificate. You can set the `SONAR_ROOT_CERT` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
 
 ## Alternatives for Java, .NET, and C/C++ projects
 

--- a/README.md
+++ b/README.md
@@ -71,7 +71,7 @@ If your sonarqube server users a self-signed certificate, you can pass a root ce
       env:
         SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
         SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
-        SONAR_ROOT_CERT: %{{ secrets.SONAR_ROOT_CERT }}
+        SONAR_ROOT_CERT: ${{ secrets.SONAR_ROOT_CERT }}
 ```
 
 You can change the analysis base directory by using the optional input `projectBaseDir` like this:

--- a/README.md
+++ b/README.md
@@ -102,7 +102,7 @@ More information about possible analysis parameters can be found in [the documen
 
 - `SONAR_TOKEN` – **Required** this is the token used to authenticate access to SonarQube. You can read more about security tokens [here](https://docs.sonarqube.org/latest/user-guide/user-token/). You can set the `SONAR_TOKEN` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
 - `SONAR_HOST_URL` – **Required** this tells the scanner where SonarQube is hosted. You can set the `SONAR_HOST_URL` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
-- `SONAR_ROOT_CERT` – This is used to pass the scanner a custom root certificate (in PEM format), that should be used for validating the sonarqube server certificate. You can set the `SONAR_ROOT_CERT` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
+- `SONAR_ROOT_CERT` – Holds an additional root certificate (in PEM format) that is used to validate the SonarQube server certificate. You can set the `SONAR_ROOT_CERT` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
 
 ## Alternatives for Java, .NET, and C/C++ projects
 

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -13,6 +13,12 @@ if [[ -z "${SONAR_HOST_URL}" ]]; then
   exit 1
 fi
 
+if [[ -n "${SONAR_ROOT_CERT}" ]]; then
+  echo "Adding custom root certificate to java certificate store"
+  echo "${SONAR_ROOT_CERT}" > /tmp/tmpcert.pem
+  keytool -keystore /etc/ssl/certs/java/cacerts -storepass changeit -noprompt -trustcacerts -importcert -alias sonarqube -file /tmp/tmpcert.pem
+fi
+
 if [[ -f "${INPUT_PROJECTBASEDIR%/}pom.xml" ]]; then
   echo "Maven project detected. You should run the goal 'org.sonarsource.scanner.maven:sonar' during build rather than using this GitHub Action."
   exit 1

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -15,6 +15,7 @@ fi
 
 if [[ -n "${SONAR_ROOT_CERT}" ]]; then
   echo "Adding custom root certificate to java certificate store"
+  rm -f /tmp/tmpcert.pem
   echo "${SONAR_ROOT_CERT}" > /tmp/tmpcert.pem
   keytool -keystore /etc/ssl/certs/java/cacerts -storepass changeit -noprompt -trustcacerts -importcert -alias sonarqube -file /tmp/tmpcert.pem
 fi

